src/gluonnlp/data/transforms.py

```diff
@@ -1221,17 +1221,29 @@ class BERTSentenceTransform:
         Tokenizer for the sentences.
     max_seq_length : int.
         Maximum sequence length of the sentences.
+    vocab : Vocab or BERTVocab
+        The vocabulary.
     pad : bool, default True
         Whether to pad the sentences to maximum length.
     pair : bool, default True
         Whether to transform sentences or sentence pairs.
     """
 
-    def __init__(self, tokenizer, max_seq_length, pad=True, pair=True):
+    def __init__(self, tokenizer, max_seq_length, vocab=None, pad=True, pair=True):
         self._tokenizer = tokenizer
         self._max_seq_length = max_seq_length
         self._pad = pad
         self._pair = pair
+        self._vocab = self._tokenizer.vocab if vocab is None else vocab
+        from ..vocab import BERTVocab
+        if isinstance(self._vocab, BERTVocab):
```

Comment:
It may be better to use duck typing here, ie. just check if the vocab defines `cls_token` and `sep_token` attributes. The only advantage of `BERTVocab` over `Vocab` currently is it's `from_sentencepiece` constructor. Once this is factored out we could delete `BERTVocab`. What do you think?